### DISCUSSION TITLE
X-Touch Mini quick reversible mod for detecting layer change

### DISCUSSION THREAD

**Discussion by bitblaster** [dterrahe/darktable#24] [Ideas]
Hi, I have the Behringer X-Touch Mini, and I did a quick and absolutely reversible mod for detecting A-B layer switch immediately from Darktable, so that the ring leds are immediately updated without having to touch other controls.
The mod is a simple hardware trick and requires just minimal soldering skills. 
I added 2x10k resistors between the layer buttons' shift register inputs and the master fader (slider/potentiometer) input. So when a layer button is pressed the fader's resistance value is sligthly changed, triggering a midi message for CC9 or CC10 (depending on the selected layer).
![immagine](https://user-images.githubusercontent.com/701527/132267576-ab266db5-043b-4386-86a3-d6fbe9826521.png)

The fader is tipically not mapped to any controls (since it's not motorized), so this spurious message doesn't interfere with the workflow.
However this mod works only when the fader slider is not at the very extreme positions: 
- if the fader is set on the highest value, the mod has no effect at all
- if the fader is set on the lowest value, the layer B will be forced. This probably could be avoided using some schottky diode in series with the resistors, however I've not tried.

Just desoldering the resistors will bring the controller back to the original behavior.

**Comment by dterrahe** [comment]
A nice hack! Thanks for letting me know.

I'm not sure how many readers will find it here, so you might want to resubmit it to the dt-docs repo, once they have started documenting Input NG. It would at least be good to have a link there to this information.

**Reply by bitblaster** [reply]
And how can I submit this? They have no Discussion panel on that repo

**Comment by dterrahe** [comment]
I would suggest filing a "Feature Request"/FR issue with either this same description or a link to another location where you may have posted it and ask that they include it. I'm not involved with that side of the project however.

**Reply by bitblaster** [reply]
ok, thanks!
